### PR TITLE
[O2B-1107] Filter on environment id

### PR DESCRIPTION
**Pull request by CollinHva**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- none

Notable changes for developers:
- API now allows filtering on get all environments

Changes made to the database:
- none

### CODE REVIEW

File: lib/usecases/environment/GetAllEnvironmentsUseCase.js [hunk 1]
```diff
@@ -35,11 +36,29 @@ class GetAllEnvironmentsUseCase {
...
+        const queryBuilder = new QueryBuilder();
+
+        if (filter) {
+            const { ids } = filter;
+            if (ids) {
+                const filters = ids.split(',').map((id) => id.trim());
+
+                // Filter should be like with only one filter
+                if (filters.length === 1) {
+                    queryBuilder.literalWhere('id LIKE :filter', { filter: `%${filters[0]}%` });
```
Comment (martinboulais) [inline lib/usecases/environment/GetAllEnvironmentsUseCase.js:51]: You should use `substring` instead

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/usecases/environment/GetAllEnvironmentsUseCase.js (hunk 1 above)
Comment (CollinHva) [reply lib/usecases/environment/GetAllEnvironmentsUseCase.js]: done

File: test/api/environments.test.js [hunk 2]
```diff
@@ -35,6 +35,15 @@ module.exports = () => {
...
+
+        it('should successfully apply filter for environments ids', async () => {
+            const response = await request(server).get('/api/environments?filter[ids]=8E4aZTjY');
+
+            expect(response.status).to.equal(200);
+            const environments = response.body.data;
+            expect(environments).to.lengthOf(1);
+            expect(environments[0].id).to.equal('8E4aZTjY');
+        });
     });
```
Comment (martinboulais) [inline test/api/environments.test.js:47]: Please add tests: without filter, with several env ids

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: test/api/environments.test.js (hunk 2 above)
Comment (CollinHva) [reply test/api/environments.test.js]: done

**Review by martinboulais — APPROVED** [review]

**Pull request merged**